src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/ParserHelpers.cs

```diff
@@ -70,5 +72,23 @@ public static bool TryParseThrowFormatException(out int bytesConsumed)
             value = default;
             return TryParseThrowFormatException(out bytesConsumed);
         }
+
+        //
+        // Enable use of ThrowHelper from TryParse() routines without introducing dozens of non-code-coveraged "value= default; bytesConsumed = 0; return false" boilerplate.
+        //
+        [DoesNotReturn]
+        [StackTraceHidden]
+        public static bool TryParseThrowFormatException<T>(ReadOnlySpan<byte> source, out T value, out int bytesConsumed) where T : struct
+        {
+            // While loop below tells C# compiler to ignore 'out not assigned' errors
```

Comment:
Couldn't we just use the new `Unsafe.SkipInit` which does the same and is fully elided?
Likewise, won't the JIT see this never returns when it hits the `ThrowFormatException` method and sees it only throws?

Reply:
Yes. Honestly the reason I didn't do that is because I tried avoiding the word "unsafe" in the code as much as possible. Maybe not a good reason?

Reply:
I think a comment saying "Bypass language initialization rules since we are throwing" will explain just fine and would make the code easier to understand